File: code/java/dailyTemperatures.java
class Solution {
    public int[] dailyTemperatures(int[] T) {
        int[] res = new int[T.length];
        int tmp;
        Stack<Integer> stack = new Stack<Integer>();
        for(int i=0;i<T.length;i++){
            while((!stack.isEmpty()) && (T[stack.peek()] < T[i])){
                tmp = stack.pop();
                res[tmp] = i - tmp;
            }
            stack.push(i);
        }
        return res;
    }
}

File: code/python/dailyTemperatures.py
class Solution:
    def dailyTemperatures(self, T: List[int]) -> List[int]:
        res = [0 for i in range(len(T))]
        stack = []
        for i in range(len(T)):
            while (stack) and T[stack[-1]] < T[i]:
                prev_index = stack.pop(-1)
                res[prev_index] = i - prev_index
            stack.append(i)

        return res
